### PR TITLE
feat: include success criteria WCAG links in injected error details dialog

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

* Adds "Success Criteria" (WCAG) info to injected dialog
* Moves `GuidanceLinks` component to `common` so we can reuse it
* Updates related dialog styling per guidance from @cheade 

Out of scope:
* Completely cleaning up the styling in the dialog beyond the scope required for implementing related visual updates
* Fixing the already-in-master issue blocking the shadow dialog from rendering

before:
image

after:
image

#### Pull request checklist

- [x] Addresses an existing issue: Addresses #519
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [x] Added screenshots/GIFs for UI changes.

### CODE REVIEW

**Comment by dbjorge** [comment]
Note to self: need to check usability with JAWS/NVDA before merging

File: src/injected/components/details-dialog.tsx [hunk 1]
```diff
@@ -253,7 +254,12 @@ export class DetailsDialog extends React.Component<DetailsDialogProps, DetailsDi
         );
     }
 
-    private renderRuleContainer(rule: DecoratedAxeNodeResult): JSX.Element {
+    private renderSectionTitle(sectionTitle: string, className?: string): JSX.Element {
+        const allClassNames = ['insights-dialog-section-title', className].filter(t => t != null);
```
Comment (smoralesd) [inline src/injected/components/details-dialog.tsx]: office fabric as a `css` function that does this for you

File: src/injected/components/details-dialog.tsx [hunk 2]
```diff
@@ -264,42 +270,58 @@ export class DetailsDialog extends React.Component<DetailsDialogProps, DetailsDi
...
+            <div className="insights-dialog-rule-name">
+                {this.renderSectionTitle('Rule name')}
+                <NewTabLink href={fixUrl(rule.helpUrl)}>{rule.ruleId}</NewTabLink>
             </div>
         );
     }
 
-    private renderFixInstructions(ruleResult: DecoratedAxeNodeResult): JSX.Element {
+    private renderSuccessCriteria(ruleGuidanceLinks: HyperlinkDefinition[]): JSX.Element {
+        if (!ruleGuidanceLinks || ruleGuidanceLinks.length === 0) {
```
Comment (smoralesd) [inline src/injected/components/details-dialog.tsx]: maybe use lodash's `isEmpty` instead?

File: src/DetailsView/components/Issues-details-pane.tsx [hunk 3]
```diff
@@ -99,6 +99,10 @@ export class IssuesDetailsPane extends React.Component<IssuesDetailsPaneProps, I
         );
     }
 
+    private renderFixInstructionsTitleElement(titleText: string, className: string): JSX.Element {
+        return <div className={className}>{titleText}</div>;
```
Comment (waabid) [inline src/DetailsView/components/Issues-details-pane.tsx]: Not sure but this doesn't look like it's being tested?

File: src/injected/styles/injected.scss [hunk 4]
```diff
@@ -389,27 +352,59 @@
...
+
+        .insights-dialog-target-button-container {
+            margin-top: 16px !important;
+
+            button {
+                padding-left: 8px !important;
+                padding-right: 8px !important;
+            }
+
+            .ms-Button-label {
```
Comment (waabid) [inline src/injected/styles/injected.scss]: Should these styles be '!important' as well?

Comment (dbjorge) [reply src/injected/styles/injected.scss]: Yeah, probably. It feels disgusting to use `!important` like this but we probably should be doing so as long as we're outside the shadow dom and competing with page styling that might use it...

File: src/injected/styles/injected.scss [hunk 5]
```diff
@@ -62,7 +63,7 @@
         width: 25%;
         max-width: 25%;
         user-select: text;
-        min-width: 550px;
+        min-width: 520px;
```
Comment (peterdur) [inline src/injected/styles/injected.scss]: Is this an intentional regression of the change in #584?

Comment (dbjorge) [reply src/injected/styles/injected.scss]: Yes, intentional; the other css changes include tightening the padding of the button contents such that the increased width is no longer necessary

**Comment by dbjorge** [comment]
Verified that it behaves as expected in jaws and nvda

**Pull request merged**